tensorflow/compiler/tf2tensorrt/segment/segment.cc

```diff
@@ -1141,6 +1143,76 @@ Status SegmentGraph(const Graph* tf_graph,
       continue;
     }
     segments->emplace_back(itr.second.property, segment_nodes);
+    effective_nodes_counts.push_back(num_effective_nodes);
+  }
+
+  // --------------------------------- Step 4 ---------------------------------
+  // If the number of segments exceeds max_engines, prune the smallest ones.
+
+  int64_t max_trt_engine_ops;
+  TF_CHECK_OK(ReadInt64FromEnvVar("TF_TRT_MAX_ALLOWED_ENGINES",
+                                  /*default_value=*/20, &max_trt_engine_ops));
+
+  if (max_trt_engine_ops <= 0) {
+    LOG(WARNING) << "The environment variable TF_TRT_MAX_ALLOWED_ENGINES is "
+                 << "<= 0. TF-TRT did not limit the number of TensorRT engines "
+                 << "created.";
+
+  } else {
+    if (segments->size() > max_trt_engine_ops) {
```

Comment:
Would you please outline this else branch into a subroutine, to improve readability?

Reply:
@bixia1 : I'm not sure to understand. You want to create a function/method just for that ?
Why ? All the other steps 1-3 are not implemented using subroutines. Why should this case be different, especially given that it's a much simpler case.

Reply:
This routine is very long, ideally, we should try to outline some functions properly. 
Your CL add one level of indentation to the original code  block, that is another reason for outlining it.

Reply:
Agreed not to do so in this PR